calico_rkt/calico_rkt.py

```diff
@@ -20,381 +20,474 @@
-    :param interface (str): iface to use
-    :param subnet (str): Subnet to allocate IP to
-    :rtype Endpoint: Endpoint created
-    """
-    # Allocate and Assign ip address through datastore_client
-    pool, ip = _assign_to_pool(subnet)
-
-    # Create Endpoint object
-    try:
-        ep = datastore_client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
-                                              container_id, [ip])
-    except AddrFormatError:
-        _log.error("This node is not configured for IPv%d. Unassigning IP "
-                   "address %s then exiting." % ip.version, ip)
-        datastore_client.unassign_address(pool, ip)
-        sys.exit(1)
-
-    # Create the veth, move into the container namespace, add the IP and
-    # set up the default routes.
-    ep.mac = ep.provision_veth(Namespace(netns_path), interface)
-    datastore_client.set_endpoint(ep)
-    return ep
-
-
-def _container_remove(hostname, orchestrator_id, container_id):
-    """
-    Remove the indicated container on this host from Calico networking
-
-    :param hostname (str): Host for enndpoint allocation
-    :param orchestrator_id (str): Specifies orchestrator ('rkt')
-    :param container_id (str):
-    """
-    # Find the endpoint ID. We need this to find any ACL rules
-    try:
-        endpoint = datastore_client.get_endpoint(hostname=hostname,
-                                                 orchestrator_id=orchestrator_id,
-                                                 workload_id=container_id)
-    except KeyError:
-        _log.error("Container %s doesn't contain any endpoints" % container_id)
-        sys.exit(1)
+    def _remove_endpoint(self):
+        """Removes the given endpoint from the Calico datastore
 
-    # Remove any IP address assignments that this endpoint has
-    for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
-        assert net.size == 1, "Only 1 address allowed per endpoint. Found in network: %s" % net
-        datastore_client.unassign_address(None, net.ip)
+        :param endpoint:
+        :return: None
+        """
+        try:
+            _log.info("Removing endpoint from the Calico datastore.")
+            self._client.remove_workload(hostname=HOSTNAME,
+                                         orchestrator_id=ORCHESTRATOR_ID,
+                                         workload_id=self.container_id)
+        except KeyError:
+            _log.error("Unable to remove workload with ID %s from datastore.",
+                       self.container_id)
+            sys.exit(1)
 
-    # Remove the endpoint
-    netns.remove_veth(endpoint.name)
+    def _provision_veth(self, endpoint):
+        """Provisions veth for given endpoint.
+
+        Uses the netns relative path passed in through CNI_NETNS_ENV and
+        interface passed in through CNI_IFNAME_ENV.
+
+        :param endpoint
+        :return Calico endpoint object
+        """
+        netns_path = os.path.abspath(os.path.join(os.getcwd(), self.netns))
```

Comment:
Let's rename `self.netns` to `self.cni_netns` to better match the CNI_NETNS environment variable naming.